docs/yul.rst

```diff
@@ -881,6 +881,12 @@ the ``dup`` and ``swap`` instructions as well as ``jump`` instructions, labels a
 | gaslimit()              |     | F | block gas limit of the current block                            |
 +-------------------------+-----+---+-----------------------------------------------------------------+
 
+The ``call*`` functions require defining an output area in memory equal to the
+``out`` plus ``outsize`` function parameters. This is not the same as the data
+returned by calling the contract, but rather the space in memory reserved for the data returned.
+The functions never touch any memory outside of this area, and return a ``0``
+("out of gas") if the area reserved is not large enough for the return data.
```

Comment:
This still needs some changes. The reserved area can be smaller than the returned data.

Reply:
We should also somehow "link" to this paragraph from the relevant lines in the table above.